src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_Scope.cs

```diff
@@ -2351,9 +2348,10 @@ into x1
             var yRef = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(name => name.Identifier.ValueText == id).ToArray();
             Assert.Equal(2, yDecl.Length);
             Assert.Equal(2, yRef.Length);
-            VerifyModelForDeclarationPattern(model, yDecl[0], yRef[1]);
-            // Should be uncommented once https://github.com/dotnet/roslyn/issues/10466 is fixed.
-            //VerifyModelForDeclarationPattern(model, yDecl[1], yRef[0]);
+            // Since the name is declared twice in the same scope,
+            // both references are to the same declaration.
+            VerifyModelForDeclarationPattern(model, yDecl[0], isShadowed: false, references: new[] { yRef[0], yRef[1] });
+            VerifyModelForDeclarationPattern(model, yDecl[1], isShadowed: true);
```

Comment:
>VerifyModelForDeclarationPattern [](start = 12, length = 32)

It looks like VerifyModelForDeclarationPatternDuplicateInSameScope should be used instead. The ```isShadowed``` case is for declarations in different scopes, can happen in switch case section.